server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -0,0 +1,771 @@
+         * else, direct to flatObjectFieldName._value subfield.
+         * @return directedSubFieldName
+         */
+        public String directSubfield() {
+            if (name().contains(DOT_SYMBOL)) {
+                String[] dotPathList = name().split("\\.");
+                return dotPathList[0] + VALUE_AND_PATH_SUFFIX;
+            } else {
+                return this.valueFieldType.name();
+            }
+        }
+
+        /**
+         * If the search key is assigned with value,
+         * the dot path was used in search query, then
+         * rewrite the searchValueString as the format "dotpath=value",
+         * @return rewriteSearchValue
+         */
+        public String rewriteValue(String searchValueString) {
+            if (!name().contains(DOT_SYMBOL)) {
+                return searchValueString;
+            } else {
+                String rewriteSearchValue = new StringBuilder().append(name()).append(EQUAL_SYMBOL).append(searchValueString).toString();
+                return rewriteSearchValue;
+            }
+
+        }
+
+        private String inputToString(Object inputValue) {
+            if (inputValue instanceof Integer) {
+                String inputToString = Integer.toString((Integer) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof Integer) {
+                String inputToString = Integer.toString((Integer) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof Float) {
+                String inputToString = Float.toString((Float) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof Boolean) {
+                String inputToString = Boolean.toString((Boolean) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof Short) {
+                String inputToString = Short.toString((Short) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof Long) {
+                String inputToString = Long.toString((Long) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof Double) {
+                String inputToString = Double.toString((Double) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof BytesRef) {
+                String inputToString = (((BytesRef) inputValue).utf8ToString());
+                return inputToString;
+            } else if (inputValue instanceof String) {
+                String inputToString = (String) inputValue;
+                return inputToString;
+            } else if (inputValue instanceof Version) {
+                String inputToString = inputValue.toString();
+                return inputToString;
+            } else {
+                // default to cast as BytesRef
+                String inputToString = ((BytesRef) inputValue).utf8ToString();
```

Comment:
This does not look right: from line 423 we know for sure this is not a `BytesRef`, the sensitive behavior  here would be:

```
return Objects.toString(inputToString);
```

Reply:
resolved. Thank you!